src/coreclr/src/vm/vars.hpp

```diff
@@ -690,12 +690,23 @@ struct ModuleIndex
 
 typedef DPTR(GSCookie) PTR_GSCookie;
 
+#ifdef _MSC_VER
+#define READONLY_ATTR
+#else
+#ifdef __APPLE__
+#define READONLY_ATTR_ARGS section("__TEXT,__const")
```

Comment:
This needs to be __DATA,__const, otherwise it would break coreclr with Mojave hardened runtime enabled on OSX. It is not allowed to change protection of executable sections to writeable there.

Reply:
I built CoreCLR on Mojave (10.14.6), we get:

```sh
$ objdump -x artifacts//bin/coreclr/OSX.x64.Debug/libcoreclr.dylib | grep s_gsCookie
0000000000000000 l    d  *UND*	_s_gsCookie
0000000000a1f4e8 l     O __TEXT,__const	_s_gsCookie
```

before and after this change. Should we still change it to `__DATA`?

Reply:
I am sorry, my mistake. My eyes have failed me when looking at the output of objdump locally. Hmm, that's strange. I need to test it with Mojave hardened runtime to see if it got somehow changed due to the compiler or whether the __TEXT,__const is actually not containing executable code.

Reply:
Ah, I have found that code is section is represented by "__TEXT,__text" and "__TEXT,__const" are "The general constant data for the executable", based on https://developer.apple.com/library/archive/documentation/Performance/Conceptual/CodeFootprint/Articles/MachOOverview.html.

So the current placement / this change is correct.